### PR TITLE
Benchmarked Serenity16 w/ gcc 4.8.5, minor bugfixes

### PR DESCRIPTION
**Pull request by cqpancoast**
Hi Joshua!

There were some bugs in here that I worked out — a couple that come to mind now are iterating from 0 to 10 instead of 1 to 10 when iterating the Coremark benchmark, and a hanging forward-slash at the end of a tarball download that prevented it from being downloaded.

Here's a thing: not all of the compiler flags work for ARM architecture. I'm not very knowledgeable about this stuff, but the ones that start with "m" just had to be removed from the flags. I hope this doesn't cause too much difference in the benchmarking between different boards. I'd add some lines to the README myself, but I'm not sure how this repository is used.

I also have a couple questions about some things that I found in here, and I'll send you a quick email about them. If there's anything that I've changed that you'd like me to not change, let me know! Again, I don't know how this repo is used, so I may have taken liberties I shouldn't have.

— Casey

### CODE REVIEW

**Comment by cqpancoast** [comment]
Also, I will be adding to this pull request by tomorrow (hopefully in an hour though) with benchmarking for Apollo 5, as well as a couple bug fixes.

File: SubmittingResults.md [hunk 1]
```diff
@@ -2,8 +2,21 @@
...
+| Whetstone             | Serenity 16: COM Express with CentOS 7 (GCC v4.8.5)      | 12857.1 (752.9) Whetstones/s    | 10     | [Run details](RunningBenchmarks.md#whetstone)|
+|-----------------------|----------------------------------------------------------|---------------------------------|--------|---------------------|
+| CoreMark              | Serenity 16: COM Express with CentOS 7 (GCC v6.2.1)      |                                 | 10     | [Run details](RunningBenchmarks.md#coremark) |
+| Dhrystone (registers) | Serenity 16: COM Express with CentOS 7 (GCC v6.2.1)      |                                 | 10     | [Run details](RunningBenchmarks.md#dhrystone-21) |
+| Dhrystone (no regs)   | Serenity 16: COM Express with CentOS 7 (GCC v6.2.1)      |                                 | 10     | [Run details](RunningBenchmarks.md#dhrystone-21) |
+| Whetstone             | Serenity 16: COM Express with CentOS 7 (GCC v6.2.1)      |                                 | 10     | [Run details](RunningBenchmarks.md#whetstone)|
```
Comment (centipeda) [inline SubmittingResults.md]: If you're planning on benchmarking the Serenity with GCC 6.2.1, do you want me to hold off on merging this pull request until these numbers are in, or do you want to just pull these in on another pull request?

**Comment by centipeda** [comment]
On the compiler flags -- what kind of architecture does the Serenity's CPU have? The flags were used on a Zynq-7000 with an ARMv7-A Cortex-A9 CPU, hence `-march=armv7a` and `-mcpu=cortex-a9`, and `-mfpu=neon-fp16` enables (I think) some ARM-specific floating-point operation acceleration. 

It would make sense for these flags not to work on non-ARM (or different ARM CPUs), but that should probably be made more clear in the README, sorry.

**Comment by cqpancoast** [comment]
Serenity's CPU has an Intel Atom processor. I was told by Tom that those -m flags wouldn't work with non-ARM processors.

File: SubmittingResults.md (hunk 1 above)
Comment (cqpancoast) [reply SubmittingResults.md]: Whatever works for you! I don't think that this data needs to be on an official repository urgently, so maybe it would be cleaner to wait until all of the data is in.

**Comment by cqpancoast** [comment]
I looked through the Whetstone script, and couldn't figure out why the time is always an integer. startsec and finisec claim to both be long — maybe it's something about the way stuff is casting?

**Comment by cqpancoast** [comment]
Here's a couple things I don't know if I've mentioned to you anywhere:
- First of all, the flags that you advise we use in RunningBenchmarks.md aren't the same flags that you use in the shell scripts that run the stuff. The flags in the shell scripts work; the ones in RB.md do not.
- Second, the flag -DNDEBUG causes the same error with both Serenity and Apollo: "-DNDEBUG is not a directory". This happened in the same way after removing the -m flags incompatible with the machines.

**Comment by centipeda** [comment]
Hmm, odd. I'll look into the flags.

DNDEBUG looks like it's used for suppressing the assertion macro, so it should be fine not to include it. I'll merge this pull request and work on these next.
